lib/Sema/CSApply.cpp

```diff
@@ -2862,8 +2834,29 @@ namespace {
           cs.getConstraintLocator(expr, ConstraintLocator::SubscriptMember);
       auto overload = solution.getOverloadChoiceIfAvailable(memberLocator);
 
-      if (overload && overload->choice.getKind() ==
-                          OverloadChoiceKind::KeyPathDynamicMemberLookup) {
+      // Handles situation where there was a solution available but it didn't
+      // have a proper overload selected from subscript call, might be because
+      // solver was allowed to return free or unresolved types, which can
+      // happen while running diagnostics on one of the expressions.
+      if (!overload) {
```

Comment:
@xedin A quick grep reveals we have no tests that emit these diagnostics.  Should this just use `getOverloadChoice`?

Reply:
Looks like there are some in d111e9b4bec632458e14e98eda99ba1bc80706da but they are crash tests so diagnostic is not verified. I have ran a couple with `-debug-constraints` and it looks like new framework is handling them well now, so I'd say let's remove this logic from here and maybe assert instead?

Reply:
Confirmed, this passes the compiler crasher suite.

Reply:
Great!